Add SVG elements & attributes

BuZZ-dEE opened the pull request:
I added a lot of SVG elements and attributes including descriptions.

BuZZ-dEE committed:
Add DTD entity

BuZZ-dEE committed:
Add SVG group

BuZZ-dEE committed:
Add shape-rendering and descriptions

BuZZ-dEE committed:
Add opacity

http://www.w3.org/TR/SVG11/masking.html#OpacityProperty
https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/opacity

BuZZ-dEE committed:
Add gitignore & merge SVG branch into master

BuZZ-dEE committed:
Add width & height

BuZZ-dEE committed:
Add shape line

BuZZ-dEE committed:
Merge branch 'svg'

BuZZ-dEE committed:
Add polyline

BuZZ-dEE committed:
Add polygon

BuZZ-dEE committed:
Add description to path element

BuZZ-dEE committed:
Add description to text element

BuZZ-dEE committed:
Add tspan element

BuZZ-dEE committed:
Add stroke attributs

BuZZ-dEE committed:
Add description properties & reorder alphabetical

BuZZ-dEE committed:
Add descriptions & fields

BuZZ-dEE committed:
Add transformations incl. descriptions

BuZZ-dEE committed:
Add clippath mask descriptions

BuZZ-dEE committed:
Add descriptions & urls

BuZZ-dEE committed:
Add image element

BuZZ-dEE committed:
Add symbol element

BuZZ-dEE committed:
Add style element

BuZZ-dEE committed:
Add filter element

BuZZ-dEE committed:
Add feGaussianBlur element

BuZZ-dEE committed:
Add title element

BuZZ-dEE committed:
Add attributes masl & clip-path

BuZZ-dEE committed:
Add viewBox attribute

BuZZ-dEE committed:
Add preserveAspectRatio attribute

BuZZ-dEE committed:
Add opacity as attribute

BuZZ-dEE committed:
Add rendering attributes

BuZZ-dEE committed:
Add color-interpolation-filters attribute

BuZZ-dEE committed:
Move to color-interpolation-filters attributes

BuZZ-dEE committed:
Add hyperlink & fe* elements

erikdahlstrom commented:
Apologies for the delay, I'll have a look at this tomorrow.

BuZZ-dEE committed:
Remove entity

Signed-off-by: Sebastian Schlatow <sebastian.schlatow@cewe.de>

erikdahlstrom commented on snippets/language-svg.cson:
```diff
@@ -1,101 +1,352 @@
+    'descriptionMoreURL': 'https://developer.mozilla.org/de/docs/Web/SVG/Element/ellipse'
+  'feColorMatrix':
+    'prefix': 'feColorMatrix'
+    'body': '<feColorMatrix in="${1:SourceGraphic}" type="${2:saturate}" values="${3:0.2}" />'
```
I think it might be better to not have the 'in' attribute here, since it'll typically do what you want without it being there.

BuZZ-dEE replied:
Removed the attribute by commit https://github.com/BuZZ-dEE/language-svg/commit/b623becf707acf29fc084a2b36f23a908dbb0e82.

erikdahlstrom commented on snippets/language-svg.cson:
```diff
@@ -1,101 +1,352 @@
+    'descriptionMoreURL': 'https://developer.mozilla.org/en-US/docs/Web/SVG/Element/feComposite'
+  'feDiffuseLighting':
+    'prefix': 'feDiffuseLighting'
+    'body': '<feDiffuseLighting in="${1:SourceGraphic}" result="${2:light}" lighting-color="${3:white}"><fePointLight x="$4" y="$5" z="$6" /></feDiffuseLighting>'
```
I don't think it's a good idea to default the 'result' attributes, as that's bound to lead to naming conflicts.

BuZZ-dEE replied:
Removed by commit https://github.com/BuZZ-dEE/language-svg/commit/d3a6972b1cc806d1df30319c2cc440cb0b76ed04.

erikdahlstrom commented on snippets/language-svg.cson:
```diff
@@ -1,101 +1,352 @@
+    'descriptionMoreURL': 'https://developer.mozilla.org/en-US/docs/Web/SVG/Element/feSpotLight'
+  'filter':
+    'prefix': 'filter'
+    'body': '<filter id="$1" x="${2:0}" y="${3:0}">$4</filter>'
```
Should have width and height too, both defaulted to "1".

BuZZ-dEE replied:
Added by commit https://github.com/BuZZ-dEE/language-svg/commit/f6a98e873983dcec55f535488659e8ac301c7f5d.

erikdahlstrom commented on snippets/language-svg.cson:
```diff
@@ -1,101 +1,352 @@
+    'descriptionMoreURL': 'https://developer.mozilla.org/en-US/docs/Web/SVG/Element/filter'
+  'group':
+    'prefix': 'g'
+    'body': '<g $1>$2</g>'
```
what's the $1 here meant for?

BuZZ-dEE replied:
That is placeholder for attributes, that the developer maybe wants to add like for example:

```
  <g id="group1" fill="red">

  </g>
```

What do you think, is it better to add some attribute or leave it out?

erikdahlstrom commented on snippets/language-svg.cson:
```diff
@@ -1,101 +1,352 @@
+    'descriptionMoreURL': 'https://developer.mozilla.org/en-US/docs/Web/SVG/Element/mask'
   'path':
     'prefix': 'p'
     'body': '<path fill="$1" stroke="$2" stroke-width="1" \n      d="$3" />\n'
```
Suggest dropping all the attributes except 'd' here.

BuZZ-dEE replied:
Dropped by https://github.com/BuZZ-dEE/language-svg/commit/174730c3248e07fef6d09f86547b5c97b170ba44

erikdahlstrom commented on snippets/language-svg.cson:
```diff
@@ -1,101 +1,352 @@
+    'descriptionMoreURL': 'https://developer.mozilla.org/de/docs/Web/SVG/Element/pattern'
+  'polygon':
+    'prefix': 'polygon'
+    'body': '<polygon points="$1,$2 $3,$4 $5,$6 $7,$8 $9,$10 $11,$12"/>'
```
wouldn't it be enough just to have a single $ here? Or perhaps two for a full coordinate. Same for polyline.

BuZZ-dEE replied:
Changed by https://github.com/BuZZ-dEE/language-svg/commit/f88ae8af9358d78a55f50cf1bda4383288d50dc4

erikdahlstrom commented on snippets/language-svg.cson:
```diff
@@ -1,101 +1,352 @@
+    'descriptionMoreURL': 'https://developer.mozilla.org/de/docs/Web/SVG/Element/script'
+  'script (internal)':
+    'prefix': 'sci'
+    'body': '<script type="${1:text/javascript}">// <![CDATA[ $2 // ]]></script>'
```
drop the type attribute, that's not necessary.

BuZZ-dEE replied:
Dropped by https://github.com/BuZZ-dEE/language-svg/commit/4667d03ac4b6a462195dd8313b1bc966eff87519

erikdahlstrom commented on snippets/language-svg.cson:
```diff
@@ -1,101 +1,352 @@
+    'descriptionMoreURL': 'https://developer.mozilla.org/de/docs/Web/SVG/Attribute/height'
+  'image-rendering':
+    'prefix': 'image-rendering'
+    'body': 'image-rendering="${1:auto}"'
```
default it to 'pixelated' instead, that seems to be what people want to use when using image-rendering. 'auto' is the default anyway, so it'd be the same as not having specified anything.

BuZZ-dEE replied:
Changed by https://github.com/BuZZ-dEE/language-svg/commit/71d2ea1458703e23d3a55de6cfee3e68e0cb8370

erikdahlstrom commented on snippets/language-svg.cson:
```diff
@@ -1,101 +1,352 @@
+    'description': 'The opacity attribute specifies the transparency of an object or of a group of objects, that is, the degree to which the background behind the element is overlaid.'
+    'descriptionMoreURL': 'https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/opacity'
+  'preserveAspectRatio':
+    'prefix': 'preserveAspectRatio'
```
suggest to make the prefix "par" instead

BuZZ-dEE replied:
Changed by https://github.com/BuZZ-dEE/language-svg/commit/bad532613bac89f9351d95d9b6bb3c678d3081e5

erikdahlstrom commented on snippets/language-svg.cson:
```diff
@@ -1,101 +1,352 @@
+    'descriptionMoreURL': 'https://developer.mozilla.org/de/docs/Web/SVG/Attribute/transform'
+  'shape-rendering':
+    'prefix': 'shape-rendering'
+    'body': 'shape-rendering="${1:auto}"'
```
suggest defaulting to 'crispEdges' instead of 'auto'.

BuZZ-dEE replied:
Changed by https://github.com/BuZZ-dEE/language-svg/commit/422c06cfa466008b80272bbd8350e844669da704

erikdahlstrom commented on snippets/language-svg.cson:
```diff
@@ -1,101 +1,352 @@
+    'descriptionMoreURL': 'https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-width'
+  'text-rendering':
+    'prefix': 'text-rendering'
+    'body': 'text-rendering="${1:auto}"'
```
suggest default 'geometricPrecision' instead of 'auto'.

BuZZ-dEE replied:
Changed by https://github.com/BuZZ-dEE/language-svg/commit/58090a6ef7db64e083cf80e6fc001b59e911742f

BuZZ-dEE committed:
Remove feColorMatrix attribute in & set link to W3C

Link to W3C SVG specification
http://www.w3.org/TR/SVG/filters.html#feColorMatrixElement

Fix for https://github.com/erikdahlstrom/language-svg/pull/2/files#r38236070

erikdahlstrom commented:
I plan to merge this later this weekend. I might do some of the minor tweaks I mentioned in a followup commit.

BuZZ-dEE committed:
Remove feComposite attribute in

Fix for https://github.com/BuZZ-dEE/language-svg/commit/a782e2efdb00ea7d3c1ad646205a987451544430#commitcomment-12945968

BuZZ-dEE committed:
Remove fiDiffuseLighting result default

Fix for https://github.com/erikdahlstrom/language-svg/pull/2#discussion_r38236553

BuZZ-dEE committed:
Add fiter attributes width & height default 1

Fix for https://github.com/erikdahlstrom/language-svg/pull/2#discussion_r38236900

BuZZ-dEE committed:
Use only path attribute d

Fix for https://github.com/erikdahlstrom/language-svg/pull/2#discussion_r38237231

BuZZ-dEE committed:
Polygon & polyline only one point $

Fix for https://github.com/erikdahlstrom/language-svg/pull/2#discussion_r38237374

BuZZ-dEE committed:
Remove script type attribute

Fix for https://github.com/erikdahlstrom/language-svg/pull/2#discussion_r38237537

BuZZ-dEE committed:
Set image-rendering default to pixelated

Fix for https://github.com/erikdahlstrom/language-svg/pull/2#discussion_r38237954

BuZZ-dEE committed:
Set preserveAspectRatio prefix to par

Fix for https://github.com/erikdahlstrom/language-svg/pull/2#discussion_r38238033

BuZZ-dEE committed:
Set shape-rendering default to crispEdges

Fix for https://github.com/erikdahlstrom/language-svg/pull/2#discussion_r38238099

BuZZ-dEE committed:
Set text-rendering default to geometricPrecision

Fix for https://github.com/erikdahlstrom/language-svg/pull/2#discussion_r38238168

BuZZ-dEE commented:
Thanks Erik. :) Let me know, if you have further suggestions,
